src/hotspot/share/memory/memoryReserver.cpp

```diff
@@ -433,13 +433,13 @@ ReservedSpace HeapReserver::Instance::try_reserve_range(char *highest_start,
   const uint64_t num_attempts_to_try   = MIN2((uint64_t)HeapSearchSteps, num_attempts_possible);
 
   const size_t stepsize = (attach_range == 0) ? // Only one try.
-    (size_t) highest_start : align_up(attach_range / num_attempts_to_try, attach_point_alignment);
+    1 : align_up(attach_range / num_attempts_to_try, attach_point_alignment);
 
-  // Try attach points from top to bottom.
-  for (char* attach_point = highest_start;
-       attach_point >= lowest_start && attach_point <= highest_start;  // Avoid wrap around.
-       attach_point -= stepsize) {
-    ReservedSpace reserved = try_reserve_memory(size, alignment, page_size, attach_point);
+  // Try reserve memory from top to bottom.
+  for (size_t offset = attach_range;
+       offset <= attach_range; // Avoid wrap around.
```

Comment:
This comment is very misleading and confusing.  This code is *relying* on underflow and the resulting
wraparound of offset to end the iteration.  That's to prevent potential pointer arithmetic underflow later.

Reply:
What about `Check for wrap around?` is that good enough?

Reply:
> What about Check for wrap around? is that good enough?

Sorry miss your review suggest just now. The comment has been updated.

Reply:
Sorry for the frequest commit...